Bugzilla/Auth/Verify/DB.pm

```diff
@@ -71,27 +71,27 @@ sub check_credentials {
   if ( Bugzilla->usage_mode == USAGE_MODE_BROWSER
     && Bugzilla->params->{password_check_on_login})
   {
-    my $pwqc = Bugzilla->passwdqc;
-    unless ($pwqc->validate_password($password)) {
-      my $reason     = $pwqc->reason;
-      my $old_reason = $user->password_change_reason;
-
-      Bugzilla->audit(sprintf "%s logged in with a weak password (reason: %s)",
-        $user->login, $reason);
-      $user->set_password_change_required(1);
-      $user->set_password_change_reason(
-        "You must change your password for the following reason: $reason");
-
-      # Remove disabled text if user was previously disabled due to inactivity
-      if ($old_reason eq 'Inactive Account') {
-        $user->set_disabledtext('');
-        $user->set_disable_mail(0);
-      }
-
-      $user->update();
+    my $check = validate_password_check($password);
+    if ($check) {
+      return {
+        failure    => AUTH_ERROR,
+        user_error => $check,
+        details    => {locked_user => $user}
+      };
```

Comment:
If I'm reading this right won't this block users from logging in with the correct password if that password doesn't match the current password requirements?